### DISCUSSION TITLE
Unauthorised Exception: 403 when registering Slash Commands

### DISCUSSION THREAD

**Discussion by samjesus8** [DSharpPlus/DSharpPlus#1489] [Support] [answered]
Someone in my Discord server is having this issue when they are starting up their Bot. When they tried to register their slash commands they are having the following error pop up in their console:

### Error Message
![image](https://user-images.githubusercontent.com/98812930/223117020-7124723d-1687-4b5a-889e-a93bcaba7599.png)

### Steps I took to help him

1. First I asked for his Bot.cs file and everything seemed normal. He set up his slash commands configuration properly and he registered his slash commands correctly
[His Pastebin](https://pastebin.com/rHST41VK)

2. I then looked at his slash commands that he is trying to register, and there are no issues with it. There are no parameters that are throwing any errors.
3. I asked him to re-generate the invite link because I know there is a similar issue that will be referenced here. I asked him to check the "Bot" and "applications.commands" when generating the invite link in the Developer Portal. It still didn't work
4. Lastly, since I have a tutorial on making a C# Discord Bot, he cloned my Example Bot repository and tried to start it up and yet again the same error pops up, even though the Example Bot is 100% working
[Example Bot Repo](https://github.com/samjesus8/YouTube-CSharp-Discord-Bot)

So even though everything looks normal, I did as much research as I could to help him, and I still cannot find an answer. What exactly is causing this error message and how do I fix it for him

**Comment by VelvetToroyashi** [comment, marked as answer]
👋🏽 Welcome to the repository! We've actually seen your tutorial, and while there are many critiques to be had about it, those shall be saved for another time.

A 403 is a permissions issue, which is indicative of either using the wrong token (which would more likely be a 401), or trying to register slash commands on a guild where it is not authorized with the `applications.commands` OAuth2 Scope, which is configured on the [Developer Dashbord](https://discord.dev).

Ensure *both* of these scopes are checked: 
![image](https://user-images.githubusercontent.com/42438262/223142126-2c69c499-73eb-4069-a4b1-01b9d0865ee1.png)

**Reply by samjesus8** [reply]
Thank you so much for for the recognition. I know my tutorial isn't 100% perfect so I'll take any feedback wherever I can.

So regarding to this issue, I told him to do this and apparently he still gets the same issue so ok, what I will do is ask him to regenerate the invite link again and maybe use a brand new empty server

I thought something was weird with his invite link so I'll try again

**Reply by VelvetToroyashi** [reply]
They're also implored to come here, or rather the [Official Library Server](https://discord.gg/DSharpPlus)

**Reply by samjesus8** [reply]
Right looks like it was the invite link and also he was confused if he needed to provide a guildID when registering commands so its all solved now

**Reply by OoLunar** [reply]
Hey that's great. While you're here Sam, could you take a look at our updated [Example Bots](https://github.com/DSharpPlus/Example-Bots) repo? It follows the best practices with the latest stable version of .NET. I think your tutorials would benefit greatly from them, and we could benefit from you suggesting us new examples! Not forced of course, but it'd be great if you could.

**Reply by samjesus8** [reply]
Sure thing, I am open to all the feedback I can get and I will make sure to analyze your Example Bots in great detail. I'm so glad my work is being recognized by the devs of the actual thing so this means a lot

**Reply by VelvetToroyashi** [reply]
A non-insignificant portion of our users have inquired about video tutorials in the past; we're glad to have [up-to-date] video tutorials for the community, but given the source-of-truth nature of this type of material, I'm sure you can imagine upholding best-practices is quintessential. 

As was mentioned before, you're definitely implored to join our [Library Server](https://discord.gg/DSharpPlus) for easier communication. We look forward to hearing from you!
